### ISSUE TITLE
Where() fails to use object property

### ISSUE THREAD

**Issue by VagyokC4** [redis/redis-om-dotnet#84]
I've been running some test on the library, and I found an issue where the Where clause only works with simple variable, and not object properties.

For example:

    var tagKey = tag.Key;
    collection.Where(c => c.Key == tag.Key).FirstOrDefault(); <---- returns null
    collection.Where(c => c.Key == tagKey).FirstOrDefault(); <---- returns the value

**Comment by slorello89** [comment]
Most likely not reading the property out correctly, will dig.

**Comment by slorello89** [comment]
Hey @VagyokC4, Does `tag` happen to be the same type as the collection?

I notice that if I do something like:

```
var anonObject = new {Name = "Steve"};
var res = collection.Where(x => x.Name == anonObject.Name).ToList();
```

works as you'd expect,

Looks like if you are using something that's a model it tries to parse it like it's looking into an embedded object.

Should be something I can remediate.

**Comment by VagyokC4** [comment]
> Looks like if you are using something that's a model it tries to parse it like it's looking into an embedded object.
> 
> Should be something I can remediate.

[Here](http://share.linqpad.net/4ed4ks.linq) is the LINQ Pad I used to test which has the model used.  I haven't tried with anonymous types, just noticed that I have to flatten my query variables out to use them and get results back.

**Comment by VagyokC4** [comment]
Also, string interpolation fails too.

    public record AnonRecord(string Name);
    var anonObject = new AnonRecord("Steve");
    var anonObjectName = anonObject.Name;
    var res1 = collection.Where(x => x.Name = anonObject.Name).ToList();           <---- FAILS
    var res2 = collection.Where(x => x.Name = $"{anonObject.Name}").ToList();      <---- FAILS
    var res3 = collection.Where(x => x.Name = $"{anonObjectName}").ToList();       <---- FAILS
    var res4 = collection.Where(x => x.Name = anonObjectName).ToList();            <---- WORKS

**Comment by slorello89** [comment]
fixed in 0.1.9
